### ISSUE TITLE
server sent ECH retry configs without client sending ECH extension

### ISSUE THREAD

**Issue by ngehrsitz** [saucesteals/utlsproxy#9]
Hi,

I am trying to intercept the traffic from a website that embeds code from `https://unpkg.com/` but that fails with the following error:
```
2024/06/07 22:43:25 [007] WARN: Cannot handshake to unpkg.com:443
%!(EXTRA *errors.errorString=tls: server sent ECH retry configs without client sending ECH extension)
```
I am using Firefox `125.0.3` as the client. I already tried toggling `network.dns.echconfig.enabled` and `network.dns.use_https_rr_as_altsvc` but that didn´t change anything. There is a specific ECH test site, that fails to load with the same error: https://defo.ie/ech-check.php
In the debugger I can see that the ECH extension is indeed missing
![image](https://github.com/saucesteals/utlsproxy/assets/45375059/64fd831e-3fbe-45fd-bba2-0695070c9ef7)
The ECH support was implemented in https://github.com/refraction-networking/utls/issues/240, but I couldn´t find an issue with that code.
Could it be that the `clientHello` is not interpreted correctly?

**Comment by dr3adx** [comment]
> Hi,
> 
> I am trying to intercept the traffic from a website that embeds code from `https://unpkg.com/` but that fails with the following error:
> 
> ```
> 2024/06/07 22:43:25 [007] WARN: Cannot handshake to unpkg.com:443
> %!(EXTRA *errors.errorString=tls: server sent ECH retry configs without client sending ECH extension)
> ```
> 
> I am using Firefox `125.0.3` as the client. I already tried toggling `network.dns.echconfig.enabled` and `network.dns.use_https_rr_as_altsvc` but that didn´t change anything. There is a specific ECH test site, that fails to load with the same error: https://defo.ie/ech-check.php In the debugger I can see that the ECH extension is indeed missing ![image](https://private-user-images.githubusercontent.com/45375059/337791281-64fd831e-3fbe-45fd-bba2-0695070c9ef7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg0Nzc5NzEsIm5iZiI6MTcxODQ3NzY3MSwicGF0aCI6Ii80NTM3NTA1OS8zMzc3OTEyODEtNjRmZDgzMWUtM2ZiZS00NWZkLWJiYTItMDY5NTA3MGM5ZWY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE1VDE4NTQzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdlZWEyZTFjMWNmMDE2OGFiNWE3NmMyM2FiZjViODU2YWM5NzI3NDc2MTExNmJjNjY1ODU3NzAxMDczN2U4ZGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3hV_ogPA_JBGRIzGM6H7GEOiVqMlevzcdth29tewztY) The ECH support was implemented in [refraction-networking/utls#240](https://github.com/refraction-networking/utls/issues/240), but I couldn´t find an issue with that code. Could it be that the `clientHello` is not interpreted correctly?

same error here, did you find a solution? throw me your discord if u have it

**Comment by ngehrsitz** [comment]
No, I didn´t find a solution. I think I will try to implement my MitM use-case with a browser plugin instead

**Comment by ngehrsitz** [comment]
I just did a quick test with `https://unpkg.com/` on 44a0906ecb39a27ba3183719cdfc423f2bfe9450 and that works fine.
But https://defo.ie/ech-check.php still refuses to work
![image](https://github.com/saucesteals/utlsproxy/assets/45375059/c2933de4-d4b4-488d-9f00-b88e12c48c89)
I can also see the same error with regular domains like `https://api.mapbox.com/` so the issue is not just limited to the test site.
